src/UI/Seller/src/app/sellers/components/locations/seller-location-edit/seller-location-edit.component.ts

```diff
@@ -0,0 +1,186 @@
+import {
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+  OnChanges,
+} from '@angular/core'
+import { FormGroup, Validators, FormControl } from '@angular/forms'
+import { Address, ListPage } from '@ordercloud/angular-sdk'
+import { MiddlewareAPIService } from '@app-seller/shared/services/middleware-api/middleware-api.service'
+import { ActivatedRoute } from '@angular/router'
+import { GeographyConfig } from '@app-seller/shared/models/supported-countries.constant'
+import {
+  ValidateCAZip,
+  ValidateUSZip,
+  ValidatePhone,
+} from '@app-seller/validators/validators'
+import { takeWhile } from 'rxjs/operators'
+import { SellerAddressService } from '../seller-address.service'
+import {
+  OrderCloudIntegrationsConversionRate,
+  HeadStartSDK,
+} from '@ordercloud/headstart-sdk'
+import { SupportedCountries } from '@app-seller/models/currency-geography.types'
+
+@Component({
+  selector: 'app-seller-location-edit',
+  templateUrl: './seller-location-edit.component.html',
+  styleUrls: ['./seller-location-edit.component.scss'],
+})
+export class SellerLocationEditComponent implements OnChanges {
+  alive = true
+  locationHasNoProducts: boolean
+  countryOptions: SupportedCountries[]
+  flag: string
+  resourceForm: FormGroup
+  zipLabel = 'Zip Code'
+  zipPlaceholder = 'Enter Zip'
+  stateLabel = 'State'
+  statePlaceholder = 'Enter State'
+  cityLabel = 'City'
+  cityPlaceholder = 'Enter City'
+  street1Label = 'Street 1'
+  street1Placeholder = 'Enter Street 1'
+  countryHasBeenSelected = false
+  currencyOptions: OrderCloudIntegrationsConversionRate[]
+  @Input()
+  set location(sellerLocation: Address) {
+    if (sellerLocation.ID) {
+      this.handleSelectedLocationChange(sellerLocation)
+    } else {
+      this.handleSelectedLocationChange(
+        this.sellerLocationService.emptyResource
+      )
+    }
+  }
+  @Input()
+  filterConfig
+  @Input()
+  suggestedAddresses: ListPage<Address>
+  @Output()
+  updateResource = new EventEmitter<any>()
+  @Output()
+  selectAddress = new EventEmitter<any>()
+  @Output()
+  canDelete = new EventEmitter<boolean>()
+
+  constructor(
+    private middleware: MiddlewareAPIService,
```

Comment:
Can you please move any methods you're using from this service into the headstart SDK? Long term I don't think this service should exist

Reply:
I updated this and the Supplier compnonent.

Reply:
Did you have to create an incrementor or did it already exist and was just not being used?

Also, did you see my note about the linting errors?

Reply:
I did and I do.  When I save the files they reformat and it offers suggestions to me.

Reply:
Those suggestions are based on best practices, so unless there's a good reason to ignore one you should strive to resolve them so there are none.

Reply:
I created a new incrementor.  The existing ones were related to specific buyers and suppliers.  I cleaned up the linter errors.

Reply:
Ah ok, so if you created a new incrementor then we'll also want to add it to the seeding process so that new orgs that are seeded have that without needing to add it themselves. Here is where you can add that:

https://github.com/ordercloud-api/headstart/blob/development/src/Middleware/src/Headstart.API/Commands/EnvironmentSeed/SeedConstants.cs#L88-L92

Reply:
Also you mentioned adding a new role right? That's also in that file towards the bottom (starting on line 247)

Reply:
Updated.